spec/controllers/groups_controller_spec.rb

```diff
@@ -479,6 +484,98 @@
+        end
+        it 'returns multiple matches' do
+          post_as instructor, :get_names, params: { course_id: course.id,
+                                                    assignment_id: assignment.id,
+                                                    assignment: assignment.id,
+                                                    term: 'zz',
+                                                    format: :json }
+
+          expect(response.parsed_body).to eq expected
+        end
+      end
+
+      describe '#assign_student_and_next' do
+        let!(:assignment) { create(:assignment_for_scanned_exam) }
+        let!(:student1) do
+          create(:student,
+                 end_user: create(:end_user,
+                                  user_name: 'c9test1', first_name: 'first', last_name: 'last', id_number: '12345'))
+        end
+        let!(:student2) do
+          create(:student,
+                 end_user: create(:end_user, user_name: 'zzz', first_name: 'zzz', last_name: 'zzz', id_number: '789'))
+        end
+        let!(:grouping1) { create :grouping, assignment: assignment }
+        let!(:grouping2) { create :grouping, assignment: assignment }
+        let!(:submission1) { create :version_used_submission, grouping: grouping1 }
+        let!(:submission2) { create :version_used_submission, grouping: grouping2 }
+        it 'assigns a student to the grouping and returns the next one' do
+          post_as instructor, :assign_student_and_next, params: { course_id: course.id,
+                                                                  assignment_id: assignment.id,
+                                                                  assignment: assignment.id,
+                                                                  names: "#{student1.first_name} #{student1.last_name}",
+                                                                  s_id: student1.id,
+                                                                  g_id: grouping1.id,
+                                                                  format: :json }
+          expect(grouping1.memberships.first.role).to eq student1
+        end
+        it 'assigns a student to the grouping and returns the next one based on names' do
+          post_as instructor, :assign_student_and_next, params: { course_id: course.id,
+                                                                  assignment_id: assignment.id,
+                                                                  assignment: assignment.id,
+                                                                  names: "#{student1.first_name} #{student1.last_name}",
+                                                                  g_id: grouping1.id,
+                                                                  format: :json }
+          expect(grouping1.memberships.first.role).to eq student1
+        end
+        it 'flashes an error if the student cannot be found' do
```

Comment:
based on above comment, also check that it renders using `head` with the appropriate status code